photonpump/conversations.py

```diff
@@ -498,6 +631,85 @@ def _fetch_page_message(self, from_event):
         return OutboundMessage(self.conversation_id, command, data, self.credential)
 
 
+
+class IterAllEvents(ReadAllEventsBehaviour, PageAllEventsBehaviour):
+    """Command class for iterating events from all events.
+
+    Args:
+        resolve_links (optional): True if eventstore should
+            automatically resolve Link Events, otherwise False.
+        required_master (optional): True if this command must be
+            sent direct to the master node, otherwise False.
+        correlation_id (optional): A unique identifer for this
+            command.
+
+    """
+
+    def __init__(
+        self,
+        from_event: int = 0,
+        batch_size: int = 500,
+        resolve_links: bool = True,
+        require_master: bool = False,
+        direction: StreamDirection = StreamDirection.Forward,
+        credentials=None,
+        conversation_id: UUID = None,
+        only_historic: bool = False
+    ):
+
+        Conversation.__init__(self, conversation_id, credentials)
+        ReadAllEventsBehaviour.__init__(
+            self, ReadAllResult, proto.ReadAllEventsCompleted
+        )
+        self.batch_size = batch_size
+        self.commit_position = from_event
+        self.only_historic = only_historic
+        self.has_first_page = False
+        self.resolve_link_tos = resolve_links
+        self.require_master = require_master
+        self.direction = direction
+        self._logger = logging.get_named_logger(IterAllEvents)
+        self.iterator = StreamingIterator(self.batch_size)
+
+        if direction == StreamDirection.Forward:
+            self.command = TcpCommand.ReadAllEventsForward
+            self.from_event = from_event or 0
+        else:
+            self.command = TcpCommand.ReadAllEventsBackward
+            self.from_event = from_event or -1
+
+    async def start(self, output: Queue):
+        await output.put(self._fetch_page_message(self.from_event))
+
+    async def success(self, result: proto.ReadAllEventsCompleted, output: Queue):
+        no_new_events = self.only_historic and result.commit_position == result.next_commit_position
+        if no_new_events:
+            self.is_complete = True
+            await self.iterator.asend(StopAsyncIteration())
+
+        if result.error == '':
+            await output.put(self._fetch_page_message(result.next_commit_position))
+
+        events = [_make_event(x) for x in result.events]
+        await self.iterator.enqueue_items(events)
+
+        if not self.has_first_page:
+            self.result.set_result(self.iterator)
+            self.has_first_page = True
+
+        if not result.error == '':
```

Comment:
This shouldn't happen, because the error path is in the `error` method, not the `success` method. Was this a bug you were dealing with, or an excess of caution?

Reply:
Oh that was there to try and detect the end of all the events but it did not work. I am  going to try add some tests by looking at your current tests. I will then remove that along with the `only_historic`